community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -0,0 +1,584 @@
+        }
+        catch ( InvalidTransactionTypeKernelException e )
+        {
+            throw new ConstraintViolationException( e.getMessage(), e );
+        }
+    }
+
+    @Override
+    public Node createNode( Label... labels )
+    {
+        try ( Statement statement = threadToTransactionBridge.get() )
+        {
+            long nodeId = statement.dataWriteOperations().nodeCreate();
+            for ( Label label : labels )
+            {
+                int labelId = statement.tokenWriteOperations().labelGetOrCreateForName( label.name() );
+                try
+                {
+                    statement.dataWriteOperations().nodeAddLabel( nodeId, labelId );
+                }
+                catch ( EntityNotFoundException e )
+                {
+                    throw new NotFoundException( "No node with id " + nodeId + " found.", e );
+                }
+            }
+            return nodeManager.newNodeProxyById( nodeId );
+        }
+        catch ( ConstraintValidationKernelException e )
+        {
+            throw new ConstraintViolationException( "Unable to add label.", e );
+        }
+        catch ( SchemaKernelException e )
+        {
+            throw new IllegalArgumentException( e );
+        }
+        catch ( InvalidTransactionTypeKernelException e )
+        {
+            throw new ConstraintViolationException( e.getMessage(), e );
+        }
+    }
+
+    @Override
+    public Node getNodeById( long id )
+    {
+        if ( id < 0 || id > MAX_NODE_ID )
+        {
+            throw new NotFoundException( format( "Node %d not found", id ),
+                    new EntityNotFoundException( EntityType.NODE, id ) );
+        }
+        try ( Statement statement = threadToTransactionBridge.get() )
+        {
+            if ( !statement.readOperations().nodeExists( id ) )
+            {
+                throw new NotFoundException( format( "Node %d not found", id ),
+                        new EntityNotFoundException( EntityType.NODE, id ) );
+            }
+
+            return nodeManager.newNodeProxyById( id );
+        }
+    }
+
+    @Override
+    public Relationship getRelationshipById( long id )
+    {
+        if ( id < 0 || id > MAX_RELATIONSHIP_ID )
+        {
+            throw new NotFoundException( format( "Relationship %d not found", id ),
+                    new EntityNotFoundException( EntityType.RELATIONSHIP, id ));
+        }
+        try ( Statement statement = threadToTransactionBridge.get() )
+        {
+            if ( !statement.readOperations().relationshipExists( id ) )
+            {
+                throw new NotFoundException( format( "Relationship %d not found", id ),
+                        new EntityNotFoundException( EntityType.RELATIONSHIP, id ));
+            }
+
+            return nodeManager.newRelationshipProxy( id );
+        }
+    }
+
+    @Override
+    public IndexManager index()
+    {
+        // TODO: txManager.assertInUnterminatedTransaction();
```

Comment:
Good question, I don't know. All the method implementations are just copied verbatim from IAGDB. That line was from a commit made by Jakub in July last year. Maybe we should add it? Not sure why it was commented out, and if we can maybe safely add it back.